app/models/search.rb

```diff
@@ -10,17 +10,29 @@ class Search < ApplicationRecord
   def self.where_terms(string)
     return all unless string
 
+    # Split each string into words by one-or-more whitespace characters
     sub_strings = string.strip.downcase.split(/\s+/).map do |str|
-      if str.start_with?("laa-")
-        clean_str = str.sub(/\Alaa-/, "laa")
-        "#{clean_str}:A"
-      elsif /\A\d+\/\d+\z/.match?(str) then "#{str}:A"
-      elsif /\A\d+{6}\z/.match?(str) then "#{str}:*"
+      # We escape with single quotes
+      str = str.gsub("'", "''")
+
+      # Is this an LAA reference or a UFN? (LAA-123ABC or 311223/001)
+      if str.start_with?("laa-") || /\A\d+\/\d+\z/.match?(str) then %('''#{str}''':A)
+
+      # Is this MAYBE a UFN? (6 digits for the first part, 9 digits for the full)
+      elsif /\A\d{6}\z/.match?(str) then "#{str}:*"
+      # Don't know what it is, but escape it and give it a lower weight anyway
```

Comment:
Nit: This is a slightly misleading comment. `B` is not a "lower" weight than `A`. Higher/lower doesn't really come into play for weights until you start messing with `ts_rank`. Rather, `A` is the "LAA reference and UFN" group, while `B` is the  "Defendant name and firm name" group. So we're saying "we know this must be either a defendant or first name, so we escape it and mark it as a query against the "B" weight"

Reply:
Yeah I wrote that before I understood what they were for better